package/MDAnalysis/visualization/streamlines.py

```diff
@@ -172,7 +174,7 @@ def produce_list_indices_point_in_polygon_this_frame(vertex_coord_list):
             list_indices_point_in_polygon.append(index_list_in_polygon)
         return list_indices_point_in_polygon
 
-    def produce_list_centroids_this_frame(list_indices_in_polygon):
+    def produce_list_centroids_this_frame(list_indices_in_polygon: Sequence[int]) -> Sequence[Union[float, str]]:
```

Comment:
```suggestion
    def produce_list_centroids_this_frame(
        list_indices_in_polygon: Iterable[int]
    ) -> List[Union[np.ndarray, str]]:
```

`list_indices_in_polygon` only needs to be an iterator, it does not need to be a sequence. The difference between the two is that an iterator only promises that you can run a for-loop once (which is what we do here), while a sequence promises you can run the for-loop more than once and you can index the object (which you do not need).

Regarding the return, you explicitly return a list so let's be explicit in the annotation, so a potential caller that explicitly needs a list is happy. Also, I think the values in the list are either the string "empty" or an array.